### DISCUSSION TITLE
How should I distribute the cluster

### DISCUSSION THREAD

**Discussion by DrBamba** [milvus-io/milvus#17262] [Q&A and General discussion]
What is considered better practice
Having a low amount of nodes with a lot of ram and CPU or a big amount of nodes with little ram and CPU

For example let's say I have a data set of 100 GB (tens millions of vectors)
Should I have 100 query nodes with 2 GB ram?

**Comment by xiaofan-luan** [comment]
> What is considered better practice Having a low amount of nodes with a lot of ram and CPU or a big amount of nodes with little ram and CPU
> 
> For example let's say I have a data set of 100 GB (tens millions of vectors) Should I have 100 query nodes with 2 GB ram?

I would highly suggest to use 4c16g or 8c32g for querynode
If each node is too small, the querycoord hard to do load balance.
If each node is too large, the failure recovery will take longer than expect
For othernode, index/data node 4c8g/4c16g should be good enough. Each coord should take no more than 1c2g/2c4g and you can deploy all node in single pod

**Reply by DrBamba** [reply]
Thank you very much!
The query nodes I should scale based on the size of the dataset
On what basis should I scale the index nodes and the data nodes?

**Reply by xiaofan-luan** [reply]
each indexnode take 4c8g would work, and indexnodes number should be mainly related to how many data is inserted and how long you can wait before you insert you data.
